src/runtime/wasm.ts

```diff
@@ -25,6 +26,65 @@ import {BiMap} from './bimap.js';
 
 type EntityTypeMap = BiMap<number, EntityType>;
 
+// Wraps a Uint8Array buffer which is automatically resized as more space is needed.
+export class DynamicBuffer {
+  private data: Uint8Array;
+  size = 0;
+
+  constructor(initialSize = 50) {
```

Comment:
Yep, arbitrary (I agonised over 50 vs 100 for ~1.6s). Since entity size is likely to be dominated by any text data contained (rather than the number of fields), not sure what we have in our current particles is going to be in any way indicative. That said, I'll just up it to 100 anyway.